src/utility/time/TimeService.cpp

```diff
@@ -129,20 +128,66 @@ unsigned long TimeService::getLocalTime()
   }
 }
 
-unsigned long TimeService::getTimeFromString(const String& timeString, const String& timeStringFormat)
+unsigned long TimeService::getTimeFromString(const String& input)
 {
-  struct tm t;
-  std::istringstream ss(timeString.c_str());
-  ss >> std::get_time(&t, timeStringFormat.c_str());
+  struct tm t =
+  {
+    0 /* tm_sec   */,
+    0 /* tm_min   */,
+    0 /* tm_hour  */,
+    0 /* tm_mday  */,
+    0 /* tm_mon   */,
+    0 /* tm_year  */,
+    0 /* tm_wday  */,
+    0 /* tm_yday  */,
+    0 /* tm_isdst */
+  };
+
+  char s_month[16];
+  int month, day, year, hour, min, sec;
+  static const char month_names[] = "JanFebMarAprMayJunJulAugSepOctNovDec";
+  static const int input_length = 20;
+  static const int input_parameters = 6;
+  unsigned long time = 0;
+
+  if(input != nullptr && input.length() == input_length) {
```

Comment:
```C++
if (input == nullptr || input.length() != input_length) {
       DEBUG_ERROR("ArduinoIoTCloudTCP::%s invalid input length", __FUNCTION__);
       return 0
}
```
I think it'd be more readable this way

Reply:
Thanks @polldo, please see 6b9602520d013a1f670abf83e9f08ffe17bb22de